api/spec/src/productcatalog/ratecards.tsp

```diff
@@ -21,7 +16,16 @@ enum RateCardType {
  * Common fields of the rate card.
  */
 @friendlyName("RateCardBase")
-model RateCardBase extends Resource {
+model RateCardBase<T extends RateCardType> {
```

Comment:
I made some changes to RateCards in https://github.com/openmeterio/openmeter/pull/1613/files#diff-b6b206b3b47ef8ef12e4ecd981d61c201a3de44ff2796ff866a0f67fb1a66e3cR33 could you take a look to see what you think?

I basically 1) did away with the Polymorphism 2) Extracted a part of it into Subscription (the part that actually gets created)